packages/block-editor/src/store/actions.js

```diff
@@ -1445,6 +1445,36 @@ export const __unstableSetEditorMode =
 			}
 		}
 
+		// TODO: Cleanup this for better performance/readability.
+		// When switching to zoom-out mode, we need to disable editing mode for all blocks except the sections.
+		const sectionsContainerClientId = select.getSectionsContainerClientId();
+		const sectionsClientIds = select.getClientIdsOfDescendants(
+			sectionsContainerClientId
+		);
+		if ( mode === 'zoom-out' ) {
+			dispatch(
+				setBlockEditingModes( [
+					[ '' /* rootClientId */, 'disabled' ],
+					[ sectionsContainerClientId, 'contentOnly' ],
+					...sectionsClientIds.map( ( clientId ) => [
+						clientId,
+						'default',
+					] ),
+				] )
+			);
+		} else {
+			const prevMode = select.__unstableGetEditorMode();
+			if ( prevMode === 'zoom-out' ) {
+				dispatch(
+					unsetBlockEditingModes( [
+						'' /* rootClientId */,
+						sectionsContainerClientId,
+						...sectionsClientIds,
+					] )
+				);
```

Comment:
I noticed we aren't keeping track of the what blocks were before, so maybe we can just use `RESET_BLOCKS` here instead.